src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,32 @@
+package duke.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends duke.task.Task {
+    private LocalDateTime by;
+
+    public Deadline(String title, LocalDateTime by) {
+        super(title);
+        this.by = by;
+    }
+
+    public Deadline(String title, LocalDateTime by, Boolean isDone) {
+        super(title, isDone);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        // can abstract these constants out into a constants file
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy hh:mm a");
+
+        return "[D] " + super.toString() + " (by: " + by.format(formatter) + ")";
+    }
+
+    public String toSaveString() {
```

Comment:
Perhaps we could differentiate this method from the previous toString further with a longer method name such as toSaveToFileString to allow developers to quickly understand its purpose